docs/reference/analysis/tokenfilters/keep-types-tokenfilter.asciidoc

```diff
@@ -1,137 +1,198 @@
 [[analysis-keep-types-tokenfilter]]
-=== Keep Types Token Filter
+=== Keep types token filter
+++++
+<titleabbrev>Keep types</titleabbrev>
+++++
 
-A token filter of type `keep_types` that only keeps tokens with a token type
-contained in a predefined set.
+Keeps or removes tokens of a specific type. For example, you can use this filter
+to change `3 quick foxes` to `quick foxes` by keeping only `<ALPHANUM>`
+(alphanumeric) tokens.
 
+[NOTE]
+.Token types
+====
+Token types are set by the <<analysis-tokenizers,tokenizer>> when converting
+characters to tokens. Token types can vary between tokenizers.
```

Comment:
Maybe worth an addition that also certain filters (e.g. synonym filter) can add tokens with a specific type (in that case of `SYNONYM` type)

Reply:
Thanks! Great point. Added a note with 5082846eeca.